lib/core/shards/usecase.ts

```diff
@@ -0,0 +1,55 @@
+import { MirrorsRepository } from '../mirrors/Repository';
+import { ShardsRepository } from './Repository';
+import { PointersRepository } from '../pointers/Repository';
+import NetworkMessageQueue from "../../server/queues/networkQueue";
+import { DELETING_FILE_MESSAGE } from "../../server/queues/messageTypes";
+import { Pointer } from "../pointers/Pointer";
+
+export class ShardsUsecase {
+  constructor(
+    private shardsRepository: ShardsRepository,
+    private mirrorsRepository: MirrorsRepository,
+    private pointersRepository: PointersRepository,
+    private networkQueue: NetworkMessageQueue
+  ) {}
+
+  async deleteShardsByIds(
+    shardIds: string[],
+    {
+      beforePointerIsDeleted,
+      version
+    }: {
+      beforePointerIsDeleted: (pointer: Pointer, version: number) => Promise<void>;
+      version: number;
+    }
+  ) {
+    const pointers = await this.pointersRepository.findByIds(shardIds);
+    for (const pointer of pointers) {
+      await beforePointerIsDeleted(pointer, version);
+      await this.pointersRepository.deleteByIds([pointer.id]);
+    }
+  }
+
+  enqueueDeleteShardMessage = async(pointer: Pointer, version: number) => {
```

Comment:
Here we have to use an arrow function or else calling this function within bucketentries usecase will make this undefined.

Reply:
it is now fixed, but previously, when it wasn't an arrow function. Calling the function from other usecases made `this` undefined. Which didn't allow us to get the repositories received from the contructor in ShardsUsecase